cgen: fixed array of option string (another C error)

### Describe the bug

C error:

### Reproduction Steps

```diff
diff --git a/vlib/wasm/tests/debug_test.v b/vlib/wasm/tests/debug_test.v
index 8d3e2e60b..14142f838 100644
--- a/vlib/wasm/tests/debug_test.v
+++ b/vlib/wasm/tests/debug_test.v
@@ -11,6 +11,7 @@ fn test_debug() {
 
        // []?string, named function parameters
        param_names := [?string('a'), 'b']
+       param_names_fixed := [?string('a'), 'b']!
 
        mut a1 := m.new_debug_function('add', func_type, param_names)
        {
```

### Expected Behavior

no C error

### Current Behavior

```
================== C compilation error (from tcc): ==============
cc: /tmp/v_1000/debug_test.01KDDDT64BH7WNQ2BYCYZHMY0K.tmp.c:10568: error: cannot convert 'struct string' to 'unsigned char'
=================================================================
Try passing `-g` when compiling, to see a .v file:line information, that correlates more with the C error.
(Alternatively, pass `-show-c-output`, to print the full C error message).
builder error: 
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V 0.4.12 e08c004

### Environment details (OS name and version, etc.)

|V full version      |V 0.4.12 bf7f43f2fe7a62502d79136c574b5b7c3376e5bf.e08c004
|:-------------------|:-------------------
|OS                  |linux, Linux version 6.13.12-200.fc41.x86_64 (mockbuild@9778ff19adb64779af61fc903691b7f0) (gcc (GCC) 14.2.1 20250110 (Red Hat 14.2.1-7), GNU ld version 2.43.1-5.fc41) #1 SMP PREEMPT_DYNAMIC Sun Apr 20 15:52:43 UTC 2025
|Processor           |4 cpus, 64bit, little endian, Intel(R) Pentium(R) Gold 7505 @ 2.00GHz
|Memory              |4.18GB/15.34GB
|                    |
|V executable        |/home/void/v/v
|V last modified time|2025-12-26 13:29:06
|                    |
|V home dir          |OK, value: /home/void/v
|VMODULES            |OK, value: /home/void/.vmodules
|VTMP                |OK, value: /tmp/v_1000
|Current working dir |OK, value: /home/void/projects/tests
|                    |
|Git version         |git version 2.49.0
|V git status        |weekly.2025.50-157-ge08c0046
|.git/config present |true
|                    |
|cc version          |cc (GCC) 14.3.1 20251022 (Red Hat 14.3.1-4)
|gcc version         |gcc (GCC) 14.3.1 20251022 (Red Hat 14.3.1-4)
|clang version       |N/A
|tcc version         |tcc version 0.9.28rc 2025-02-13 HEAD@f8bd136d (x86_64 Linux)
|tcc git status      |thirdparty-linux-amd64 696c1d84
|emcc version        |N/A
|glibc version       |ldd (GNU libc) 2.40
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/array.v
+++ b/vlib/v/gen/c/array.v
@@ -175,7 +175,11 @@ fn (mut g Gen) fixed_array_init(node ast.ArrayInit, array_type Type, var_name st
 				} else {
 					node.elem_type
 				}
-				g.expr_with_cast(expr, expr_type, node.elem_type)
+				if node.elem_type.has_flag(.option) {
+					g.expr_with_opt(expr, expr_type, node.elem_type)
+				} else {
+					g.expr_with_cast(expr, expr_type, node.elem_type)
+				}
 			}
 			g.add_commas_and_prevent_long_lines(i, nelen)
 		}

--- a/vlib/v/tests/options/option_array_init_mix_test.v
+++ b/vlib/v/tests/options/option_array_init_mix_test.v
@@ -1,0 +1,9 @@
+fn test_main() {
+	a := [?string('a'), 'b']!
+	println(a)
+	assert '${a}' == "[Option('a'), Option('b')]"
+
+	b := [?int(1), 7]!
+	println(b)
+	assert '${b}' == '[Option(1), Option(7)]'
+}

